r/configure.win

```diff
@@ -104,6 +105,10 @@ function configure_dev() {
     PKG_CFLAGS="$PKG_CFLAGS -DARROW_R_WITH_S3"
   fi
 
+  if [ $(cmake_option ARROW_GCS) -eq 1 ]; then
+    PKG_CFLAGS="$PKG_CFLAGS -DARROW_R_WITH_GCS -DCURL_STATICLIB"
+  fi
```

Comment:
I think we should be able to let pkg-config handle adding the flags and libraries for libcurl:

```suggestion
  if [ $(cmake_option ARROW_GCS) -eq 1 ]; then
    PKG_CFLAGS="$PKG_CFLAGS -DARROW_R_WITH_GCS"
    PKG_CONFIG_PACKAGES="$PKG_CONFIG_PACKAGES libcurl"
  fi
```

Reply:
Nevermind this might not help if libcurl is already statically linked in the arrow binaries.